include/beast/http/file_body_linux.hpp

```diff
@@ -0,0 +1,256 @@
+//
+// Copyright (c) 2013-2017 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BEAST_HTTP_FILE_BODY_LINUX_HPP
+#define BEAST_HTTP_FILE_BODY_LINUX_HPP
+
+#ifdef __linux__
+# define BEAST_NO_LINUX_FILE 0
+#else
+# define BEAST_NO_LINUX_FILE 1
+#endif
```

Comment:
I've added the extra `#ifdef` check. I think a proper feature test should be left until this gets merged properly -- it might really be better as a "posix" rather than "linux" file body, as the implementation will just differ in the write specialisations.